src/sagemaker/workflow/steps.py

```diff
@@ -394,24 +395,40 @@ def __init__(
 
         Args:
             name (str): The name of the `CreateModelStep`.
+            step_args (dict): The arguments for the `CreateModelStep` definition (default: None).
             model (Model or PipelineModel): A `sagemaker.model.Model`
-                or `sagemaker.pipeline.PipelineModel` instance.
+                or `sagemaker.pipeline.PipelineModel` instance (default: None).
             inputs (CreateModelInput): A `sagemaker.inputs.CreateModelInput` instance.
-                Defaults to `None`.
+                (default: None).
             depends_on (List[str] or List[Step]): A list of `Step` names or `Step` instances
-                this `sagemaker.workflow.steps.CreateModelStep` depends on.
-            retry_policies (List[RetryPolicy]):  A list of retry policies.
-            display_name (str): The display name of the `CreateModelStep`.
-            description (str): The description of the `CreateModelStep`.
+                this `sagemaker.workflow.steps.CreateModelStep` depends on (default: None).
+            retry_policies (List[RetryPolicy]):  A list of retry policies (default: None).
+            display_name (str): The display name of the `CreateModelStep` (default: None).
+            description (str): The description of the `CreateModelStep` (default: None).
         """
         super(CreateModelStep, self).__init__(
             name, StepTypeEnum.CREATE_MODEL, display_name, description, depends_on, retry_policies
         )
+        if not (step_args is None) ^ (model is None):
+            raise ValueError(
+                "step_args and model are mutually exclusive. Either of them should be provided."
+            )
+
+        self.step_args = step_args
         self.model = model
         self.inputs = inputs or CreateModelInput()
 
         self._properties = Properties(path=f"Steps.{name}", shape_name="DescribeModelOutput")
 
+        if not self.step_args:
+            warnings.warn(
+                (
+                    "We are deprecating the instantiation of CreateModelStep using "
+                    "bunch of arguments. Instead, the new interface uses step_args."
```

Comment:
We are deprecating the instantiation of CreateModelStep using "model". Instead, the new interface uses step_args."

Reply:
Updated in the new commit